TACC/protx-dashboard
# Task/COOKS-200: add container for protx backend

## Overview: ##

Add container for protx that runs along CEPv2.

See README.md for getting started



## Related Jira tickets: ##

* [COOKS-200](https://jira.tacc.utexas.edu/browse/COOKS-200) substask of [COOKS-21](https://jira.tacc.utexas.edu/browse/COOKS-21)

## Summary of Changes: ##
* Docker-compose and other configuration stuff for local development
* Add existing protx backend from https://github.com/TACC/protx
* Add unit tests
* Added commit with previous repo's coauthors [93835f7](https://github.com/TACC/protx-dashboard/pull/1/commits/93835f769fe1ab190c0772f90b131d0d52c02849)

## Testing Steps: ##
0. See README for getting started
1. Go to `https://cep.dev/workbench`
2. Go to `https://cep.dev/protx/api/maltreatment`
3. Confirm that you need to be logged in (with setupComplete) by logging out and going to `https://cep.dev/protx/api/maltreatment`. Confirm that you get a 403
3. Get static geospatial files (like `wget cep.dev/data-static/Texas_State_Boundary.geojson` or `wget cep.dev/data-static/vector/county/2019/6/14/25.pbf`)

## Notes: ##

TODO
- [x] revert 5ddb7c2d93a5152bbf52d0a7c96f4d9c555e84f0 once clarify why onboarding steps aren't running correctly.
- [x] flake8 
- [x] gihub actions for flake8
- [x] add setting_secrets to stache (It is just using normal CEP secrets https://stache.utexas.edu/entry/bedc97190d3a907cb44488785440595c)
- [x] add commit to add history of other contributors
- [x] pin python requirements in `requirements.txt` or in poetry
- [x] `ConnectionError at /remote/login/` failure
- [x] use `protx_geospatial_data` (i.e. taccwma/protx-geospatial:latest)
- [x] update backend to match what is in development at https://github.com/TACC/protx and specifically [task/COOKS-183-new-simplified-maltreatment-chart](https://github.com/TACC/protx/pull/94)


TODO later (i.e. make jira issues)
- [x] add jira issue for db fixtures for unit testing  (nice-to-have: [COOKS-254](https://jira.tacc.utexas.edu/browse/COOKS-254)
- [x] add jira issue for unified access of our two dbs (i.e. proper closing etc) (nice-to-have: [COOKS-255](https://jira.tacc.utexas.edu/browse/COOKS-255))
- [x] add jira issue for frontend finalization (i.e. template etc) for prod/staging deployment https://github.com/TACC/protx-dashboard/pull/2#discussion_r849906776. (**MUST BE DONE**: [COOKS-256](https://jira.tacc.utexas.edu/browse/COOKS-256))
- [x] add jira issue to route protx dashboard route to workbench/onboarding if user is not setupComplete (see https://github.com/TACC/protx-dashboard/pull/2#discussion_r849934311)  (**MUST BE DONE**: [COOKS-257](https://jira.tacc.utexas.edu/browse/COOKS-257))
- [x] add jira issue issue to port recent changes from forked repo (**MUST BE DONE**: [COOKS-258](https://jira.tacc.utexas.edu/browse/COOKS-258))
- [x] add jira issue  for globe icon not being shown.  (**MUST BE DONE**: [COOKS-259](https://jira.tacc.utexas.edu/browse/COOKS-259))
- [x] add jira issue  for github actions workflow for frontend (unit testing and linting).  (nice-to-have: [COOKS-260](https://jira.tacc.utexas.edu/browse/COOKS-260)

Gonna add following when merging:

```
Co-authored-by: John Gentle <taoteg@gmail.com>
Co-authored-by: Kelly Pierce <kellyannepierce@gmail.com>
Co-authored-by: Ian Park <contact@duckonomy.com>
Co-authored-by: Jake Rosenberg <jrosenberg@tacc.utexas.edu>
```
